HistoricalDataFetcher.Classes/Utilities/ApiRequest.cs

```diff
@@ -0,0 +1,172 @@
+            _urlBase = $"https://{_httpHost}/api";
+
+            return await AuthenticateAsync();
+        }
+
+        /// <summary>
+        /// Authenticates and runs the URL with the acceptHeader
+        /// </summary>
+        /// <param name="url">Complete URL</param>
+        /// <param name="acceptHeader">header used to make the call</param>
+        /// <returns>Endpoint Result</returns>
+        public static async Task<string> RunEndpointCallAsync(string url, string acceptHeader)
+        {
+            if (_accessTokenInstance == null)
+            {
+                await AuthenticateAsync();
+            }
+
+            var result = _cache.Get(url);
+            return !string.IsNullOrEmpty(result) ? result : await RunEndpointHelperAsync(url, acceptHeader);
+        }
+
+        /// <summary>
+        /// Runs the URL using the required header info
+        /// </summary>
+        /// <param name="url">Complete URL</param>
+        /// <param name="acceptHeader">header used to make the call</param>
+        /// <returns>EndPoint result</returns>
+        private static async Task<string> RunEndpointHelperAsync(string url, string acceptHeader)
+        {
+            string rawJsonToReturn = string.Empty;
+            var uri = new Uri(url);
+
+            using (var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true,
+                AllowAutoRedirect = false
+            })
+            using (var httpClient = new HttpClient(handler))
+            {
+                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _accessTokenInstance.Access_Token);
+                var response = await httpClient.GetAsync(uri);
+                
+                if (response.StatusCode == HttpStatusCode.Moved || response.StatusCode == HttpStatusCode.Redirect)
+                {
+                    return await RunEndpointHelperAsync(response.Headers.Location.ToString(), acceptHeader);
+                }
+                else
+                {
+                    rawJsonToReturn = await response.Content.ReadAsStringAsync();
+                }
+            }
+
+            _cache.Add(url, rawJsonToReturn);
+            _cache.Persist();
+
+            return rawJsonToReturn ?? string.Empty;
+        }
+
+        /// <summary>
+        /// Authenticate access using the host location
+        /// </summary>
+        /// <returns>bool: True = success</returns>
+        private static async Task<bool> AuthenticateAsync()
+        {
+            var body = $"{{\"username\": \"{_username}\",\"password\": \"{_password}\"}}";
+
+            var uri = new Uri($"{UrlBase}/api/authentication/login");
+    
+            using (var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true
```

Comment:
let's find all of these and fix them.

The only acceptable implementation is to examine the error, and if it's related to self signed certificate, AND if the customer passed a flag saying that they are okay with insecure mode, then you can return true. See how I did it in objectjs and alarmjs and nodekit and curl-scripts.

I allow for them to specify a cert saying they trust it, or they can pass --insecure and I warn them not to do that in product. We certainly don't want to hardcode a solution like this.

Reply:
However see my other comment on just telling a consumer how to setup their OS to trust their self-signed cert.

Reply:
This has been address via the HandleCertificateError method.